### PR TITLE
Feature/issue 9/create backend structure

### LINKED ISSUE #9: Create Backend Structure
Image

### PR DESCRIPTION
**Pull request by huy1707developer**
More detail in #9

### CODE REVIEW

**Comment by nnh53** [comment]
@huy1707developer pls revert your commit or change your commit name follow by the convention in QUY_TẮC_COMMIT.md

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import swaggerUi from 'swagger-ui-express'
+import { config } from 'dotenv'
+import databaseService from './module/database/database.service'
+import express from 'express'
+import cors from 'cors'
+import YAML from 'yaml'
```
Comment (nnh53) [inline src/index.ts:6]: pls remove unused import

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import fs from 'fs'
+import path from 'path'
+import swaggerJSDoc from 'swagger-jsdoc'
+
+config()
+//cors
+const corsOptions = {
+  origin: 'http://localhost:4000' || 'http://localhost:5173',
```
Comment (nnh53) [inline src/index.ts:14]: make this to ENV

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+//app handler
+app.use(express.json())
+//route
+
+//database connect
+databaseService.connect()
+
+//Port
+app.listen(PORT, () => {
+  console.log(`server này đang chạy trên port:::${PORT}`)
```
Comment (nnh53) [inline src/index.ts:32]: just one :

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      {
+        "arrowParens": "always",
+        "semi": false,
+        "trailingComma": "none",
+        "tabWidth": 2,
+        "endOfLine": "auto",
+        "useTabs": false,
+        "singleQuote": true,
+        "printWidth": 120,
+        "jsxSingleQuote": true
```
Comment (nnh53) [inline .eslintrc:20]: are you sure ?? we just use ts

File: .gitattributes [hunk 5]
```diff
@@ -0,0 +1,4 @@
+/.yarn/**            linguist-vendored
+/.yarn/releases/*    binary
+/.yarn/plugins/**/*  binary
+/.pnp.*              binary linguist-generated
```
Comment (nnh53) [inline .gitattributes:4]: what is this file for

File: .prettierrc [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  "arrowParens": "always",
+  "semi": false,
```
Comment (nnh53) [inline .prettierrc:3]: revert the tab

File: package-lock.json [hunk 7]
```diff
@@ -0,0 +1,6062 @@
+{
+  "name": "backend",
```
Comment (nnh53) [inline package-lock.json:2]: we use yarn-lock not package-lock

File: tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
+{
+  "compilerOptions": {
+    "module": "CommonJS", // Quy định output module được sử dụng
```
Comment (nnh53) [inline tsconfig.json:3]: remove unnecessary comment, only leave what you need

File: .yarnrc.yml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+nodeLinker: node-modules
+
```
Comment (nnh53) [inline .yarnrc.yml:2]: what is this file for

**Review by nnh53 — COMMENTED** [review]
fix alot pls

**Review by nnh53 — CHANGES_REQUESTED** [review]
test request change, and please remove the learn-swagger file

File: package.json [hunk 10]
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "dreame-backend",
```
Comment (nnh53) [inline package.json:2]: fix to this "dreame-clone-training-be", just to be carefull @@

File: package.json [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+  "packageManager": "yarn@4.1.1",
```
Comment (nnh53) [inline package.json:3]: image.png, this is just a Experimental feature

File: package.json [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+    "dev": "yarn run nodemon",
+    "build": "rimraf ./dist && tsc && tsc-alias",
+    "start": "node dist/src/index.js --production",
+    "lint": "eslint .",
+    "lint:fix": "eslint . --fix",
+    "prettier": "prettier --check .",
+    "prettier:fix": "prettier --write ."
+  },
+  "author": "",
+  "license": "MIT",
```
Comment (nnh53) [inline package.json:20]: "author": "",
"license": "MIT"
remove these 2 and at this at the top
"private": true,

**Comment by nnh53** [comment]
@CiscoTran can you help him

**Comment by huy1707developer** [comment]
@lcaohoanq, can you review for me?

**Comment by lcaohoanq** [comment]
> @lcaohoanq, can you review for me?

No, i can't the reviewer is only
image

**Comment by nnh53** [comment]
@huy1707developer it your PR, you can add more people to review to, or is it only me can add reviewer ??

**Review by lcaohoanq — DISMISSED** [review]
@huy1707developer 
I've done review your PR

**Review by lcaohoanq — APPROVED** [review]

**Review by lcaohoanq — APPROVED** [review]

**Review by lcaohoanq — DISMISSED** [review]

File: .eslintrc [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  "plugins": ["@typescript-eslint", "prettier"],
+  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended", "eslint-config-prettier", "prettier"],
+  "rules": {
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/no-unused-vars": "off",
+    "prettier/prettier": [
+      "warn",
+      {
+        "arrowParens": "always",
+        "semi": false,
```
Comment (minhhy2801) [inline .eslintrc:13]:
```suggestion
        "semi": true,
```

File: .eslintrc [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+        "trailingComma": "none",
```
Comment (minhhy2801) [inline .eslintrc:14]: should have

File: .gitignore [hunk 15]
```diff
@@ -0,0 +1,166 @@
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
```
Comment (minhhy2801) [inline .gitignore:6]: You didn't use them

File: src/index.ts [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+
+config()
+const app = express()
+const PORT = process.env.PORT || 4000
+//cors
+const corsOptions = {
+  origin: `http://localhost:${PORT}`,
+  credentials: true, //access-control-allow-credentials:true
+  optionSuccessStatus: 200
+}
```
Comment (minhhy2801) [inline src/index.ts:13]: add type for them

File: tsconfig.json [hunk 17]
```diff
@@ -0,0 +1,23 @@
+{
+  "compilerOptions": {
+    "module": "CommonJS",
```
Comment (minhhy2801) [inline tsconfig.json:3]: why do you pick CommonJS here?

**Review by nnh53 — DISMISSED** [review]

**Review by nnh53 — APPROVED** [review]

**Pull request merged**